channel/channel.go

```diff
@@ -328,37 +334,36 @@ func (c *Channel) SignAndAddState(s state.State, sk *[]byte) (state.SignedState,
 
 // UpdateWithChainEvent mutates the receiver with the supplied chain event, replacing the relevant data fields.
 func (c *Channel) UpdateWithChainEvent(event chainservice.Event) (*Channel, error) {
-	if event.BlockNum() > c.LastChainUpdate.BlockNum ||
-		(event.BlockNum() == c.LastChainUpdate.BlockNum && event.TxIndex() > c.LastChainUpdate.TxIndex) {
-		// Process event
-		switch e := event.(type) {
-		case chainservice.AllocationUpdatedEvent:
-			c.OnChain.Holdings[e.AssetAddress] = e.AssetAmount
-			// TODO: update OnChain.StateHash and OnChain.Outcome
-		case chainservice.DepositedEvent:
-			c.OnChain.Holdings[e.Asset] = e.NowHeld
-		case chainservice.ConcludedEvent:
-			break // TODO: update OnChain.StateHash and OnChain.Outcome
-		case chainservice.ChallengeRegisteredEvent:
-			h, err := e.StateHash(c.FixedPart)
-			if err != nil {
-				return nil, err
-			}
-			c.OnChain.StateHash = h
-			c.OnChain.Outcome = e.Outcome()
-			ss, err := e.SignedState(c.FixedPart)
-			if err != nil {
-				return nil, err
-			}
-			c.AddSignedState(ss)
-		default:
-			return &Channel{}, fmt.Errorf("channel %+v cannot handle event %+v", c, event)
+	if !c.isNewChainEvent(event) {
+		return nil, fmt.Errorf("chain event older than channel's last update")
+	}
```

Comment:
I believe this pattern is more idiomatic https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88 .